packages/babel-helper-import-to-platform-api/src/index.ts

```diff
@@ -0,0 +1,228 @@
+    toCJS: c = toCommonJS,
+  }: {
+    web: boolean;
+    node: boolean;
+    webIMR?: boolean;
+    nodeIMR?: boolean;
+    toCJS?: boolean;
+    preferSync?: boolean;
+  }) => +w + (+n << 1) + (+wI << 2) + (+nI << 3) + (+c << 4);
+
+  switch (
+    p({
+      web: needsWebSupport,
+      node: needsNodeSupport,
+      webIMR: webSupportsIMR,
+      nodeIMR: nodeSupportsIMR,
+      toCJS: toCommonJS,
+    })
+  ) {
+    case p({ web: true, node: true }):
+      buildFetchAsync = specifier => {
+        const web = transformers.webFetch(
+          t.callExpression(t.identifier("fetch"), [
+            (webSupportsIMR ? imr : imrWithFallback)(t.cloneNode(specifier)),
+          ]),
+        );
+        const node = nodeSupportsIMR
+          ? template.expression.ast`
+                import("fs").then(
+                  fs => fs.promises.readFile(new URL(${imr(specifier)}))
+                ).then(${transformers.nodeFsAsync()})
+              `
+          : template.expression.ast`
+                Promise.all([import("fs"), import("module")])
+                  .then(([fs, module]) =>
+                    fs.promises.readFile(
+                      module.createRequire(import.meta.url)
+                        .resolve(${specifier})
+                    )
+                  )
+                  .then(${transformers.nodeFsAsync()})
+              `;
+
+        return template.expression.ast`
+            typeof process === "object" && process.versions?.node
+              ? ${node}
+              : ${web}
+          `;
+      };
+      break;
+    case p({ web: true, node: true, webIMR: false, nodeIMR: true }):
+      buildFetchAsync = specifier => template.expression.ast`
+          typeof process === "object" && process.versions?.node
+            ? import("fs").then(fs =>
+                ${transformers.nodeFsAsync()}(fs.readFileSync(
+                  new URL(${imr(specifier)})
+                ))
+              )
+            : ${transformers.webFetch(
+              t.callExpression(t.identifier("fetch"), [
+                imrWithFallback(specifier),
+              ]),
+            )}
+        `;
+      break;
+    case p({ web: true, node: false, webIMR: true }):
+      buildFetchAsync = specifier =>
+        transformers.webFetch(
+          t.callExpression(t.identifier("fetch"), [imr(specifier)]),
+        );
+      break;
+    case p({ web: true, node: false, webIMR: false }):
+      buildFetchAsync = specifier =>
+        transformers.webFetch(
+          t.callExpression(t.identifier("fetch"), [imrWithFallback(specifier)]),
+        );
+      break;
+    case p({ web: false, node: true, toCJS: true }):
+      buildFetchSync = specifier =>
+        transformers.nodeFsSync(template.expression.ast`
+            require("fs").readFileSync(require.resolve(${specifier}))
+          `);
+      buildFetchAsync = specifier => template.expression.ast`
+            require("fs").promises.readFile(require.resolve(${specifier}))
```

Comment:
The fs Promises API is available on Node.js 10. If we would like to support older versions, we should manually promisify the callback-style `fs.readFile` call. Ideally we can output the most compatible form based on the input target.